packages/flutter/lib/src/rendering/sliver_persistent_header.dart

```diff
@@ -738,13 +738,17 @@ abstract class RenderSliverFloatingPersistentHeader extends RenderSliverPersiste
     // A stretch header can have a bigger childExtent than maxExtent.
     final double effectiveMaxExtent = math.max(childExtent, maxExtent);
 
-    targetExtent = targetExtent.clamp(
-        showOnScreen.minShowOnScreenExtent,
-        showOnScreen.maxShowOnScreenExtent,
-      )
-      // Clamp the value back to the valid range after applying additional
-      // constraints. Contracting is not allowed.
-      .clamp(childExtent, effectiveMaxExtent);
+    targetExtent = clampDouble(
+        clampDouble(
+          targetExtent,
+          showOnScreen.minShowOnScreenExtent,
+          showOnScreen.maxShowOnScreenExtent,
+        )
+        // Clamp the value back to the valid range after applying additional
+        // constraints. Contracting is not allowed.
+        ,
```

Comment:
this formatting seems a bit off, maybe move the `,` up to the closing paren above?

Reply:
done